src/main/java/io/vertx/ext/stomp/impl/DefaultStompHandler.java

```diff
@@ -0,0 +1,563 @@
+        handleDisconnect(frame, connection);
+        break;
+      case PING:
+        // We received a ping frame, we do nothing.
+        break;
+    }
+  }
+
+  private void handleAck(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = ackHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleNack(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = nackHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleBegin(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = beginHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleAbort(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = abortHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleCommit(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = commitHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleSubscribe(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = subscribeHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleUnsubscribe(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = unsubscribeHandler;
+    }
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleSend(Frame frame, StompServerConnection connection) {
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = sendHandler;
+    }
+
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+  }
+
+  private void handleConnect(Frame frame, StompServerConnection connection) {
+
+    ServerFrameHandler handler;
+    synchronized (this) {
+      handler = connectHandler;
+    }
+
+    if (handler != null) {
+      handler.onFrame(frame, connection);
+    }
+    // Compute heartbeat, and register pinger and ponger
+    long ping = Frame.Heartbeat.computePingPeriod(
+        Frame.Heartbeat.parse(frame.getHeader(Frame.HEARTBEAT)),
+        Frame.Heartbeat.create(connection.server().getOptions().getHeartbeat()));
+    long pong = Frame.Heartbeat.computePongPeriod(
+        Frame.Heartbeat.parse(frame.getHeader(Frame.HEARTBEAT)),
+        Frame.Heartbeat.create(connection.server().getOptions().getHeartbeat()));
+    if (ping > 0) {
```

Comment:
I think it would be more efficient to have a single periodic timer that sent the pings and checked the lastactivity